### ISSUE TITLE
java.net.URL fields throw exception

### ISSUE THREAD

**Issue by johngluck65** [oshoukry/openpojo#74]
This is similar to issue https://github.com/oshoukry/openpojo/issues/72.  So I'm wondering if it makes more sense, rather than simply adding classes one by one, to allow some method for extending the RandomGenerator.  I can follow your example for the fix to issue #72 and submit a PR but I'm not certain when I'll get around to it.  That said, I can see a situation where I have a field that is part of a private library and it wouldn't make sense to extend the project for that.  I'm not sure how that would work and this is really a feature request, I guess.  I'm just wondering what your thoughts and plans are.  I would appreciate your advice on how to proceed on java.net.URL.

**Comment by oshoukry** [comment]
There is a simple solution, OpenPojo tries to do its best to create an instance of the requested object, but if that object has specific requirements, a RandomGenerator can be created to handle it.

Once you create a RandomGenerator that implements that interface, you can register it through the RandomFactory as so:

```Java
import com.openpojo.random.RandomGenerator;

[...] // Class definition

@BeforeClass
public static void onlyOnce() {
  MyRandomGenerator generator = new MyRandomGenerator();

  // This registration will be there for all unit tests, not just this class
  RandomFactory.addRandomGenerator(generator);
}

[...]
```

see https://github.com/oshoukry/openpojo/blob/master/src/main/java/com/openpojo/random/RandomFactory.java

**Comment by oshoukry** [comment]
For basic classes like UUID and URL, I feel they are helpful to anyone using the framework so I am happy to add them in the framework, thank you for bringing them up, I'll add them for the next release.

**Comment by johngluck65** [comment]
Okay, great.  That's very helpful.  Thanks for the information

**Comment by johngluck65** [comment]
Just a suggestion, I think it would be helpful to add your very clear example above to the wiki documentation.  Although I was just on the brink of figuring out how to do this when you answered my question, it wasn't entirely self evident without some digging.

**Comment by oshoukry** [comment]
Thank you for your continued support and feature requests.
This issue has been fixed, please download OpenPojo-0.8.3.

I will be adding the wiki tutorial page to how to extend the RandomGenerator interface with custom generators.
